File: src/Modules/Contracts/Device.Contract/DTOs/ModuleDto.cs
using System;
using SharedKernel.Domain;

namespace Device.Contract.DTOs;

public sealed record ModuleDto(
      Guid Guid = default,
      string Name="",
      string Fw="",
      string SerialNumber="",
      short Port=0,
      short Address=0,
      string Mac="",
      string Model="",
      string Type="",
      Guid DeviceGuid=default,
      int LocationId=0,
      bool IsActive=true,
      bool IsDefault=false
      ) : BaseDtoEntity(Guid,LocationId,Type,IsActive,IsDefault);

File: src/Web/src/model/Module/ModuleDto.ts
export interface ModuleDto
{
    id:number,
    componentId:number;
    deviceComponentId:number;
    name:string,
    fw:string,
    serialNumber:string,
    port:number,
    address:number,
    mac:string,
    model:string,
    locationId:number;
    type:string;
    isActive:boolean;
}